myokit/_model_api.py

```diff
@@ -1659,6 +1659,276 @@ def has_warnings(self):
+           other.
+        3. By assuming that variables with the same qualified names map onto
+           each other. This rule is only applied if ``allow_name_mapping`` is
+           set to ``True``.
+
+        Arguments:
+
+        ``external_component``
+            A :class:`myokit.Component` from another model.
+        ``new_name``
+            An optional new name for the imported component.
+        ``var_map``
+            An optional dict mapping variables in the source model to
+            variables from this model (with variables specified as objects or
+            as fully qualified names).
+        ``allow_name_mapping``
+            Set this to ``True`` to allow mapping based on fully qualified
+            names, e.g. a variable ``x.y`` in the source model will be mapped
+            to ab variable ``x.y`` in the target model.
+        ``convert_units``
+            Set this to ``True`` to convert the units of any mapped variables,
+            if required and possible.
+
+        If any variables cannot be mapped, a :class:`myokit.WellMappedError`
+        will be raised. A :class:`myokit.DuplicateNameError` will be raised if
+        the components name or ``new_name`` clashes with an existing component
+        name. If unit conversion is enabled and variables with incompattible
+        units are mapped onto each other, a
+        class:`myokit.IncompatibleUnitsError` will be raised.
+        """
+
+        # Check component is not from this model, is a component, etc.
+        if not isinstance(external_component, myokit.Component):
+            raise TypeError(
+                'Method import_component() expects a myokit.Component'
+            )
+        if external_component.has_ancestor(self):
+            raise myokit.IntegrityError(
+                '<' + external_component.name() +
+                '> already has this model as an ancestor'
+            )
+
+        # Check if new name is provided, or else check that name doesn't clash
+        if new_name is None:
+            new_name = external_component.name()
+        if self.has_component(new_name):
+            raise myokit.DuplicateName(
+                'This model already has component with the name ' + new_name
+            )
+
+        # Create mapping for vars, and get unit conversion factors if required.
+
+        ext_model_copy = external_component.model().clone()
+        external_component_copy = ext_model_copy[external_component.qname()]
+
+        full_var_map = {}
+        relevant_vars = external_component_copy.variables()
+        for var in external_component_copy.variables():
+            dependent_vars = var.refs_to(state_refs=False)
+            dependent_states = var.refs_to(state_refs=True)
+            relevant_vars = (
+                list(relevant_vars) +
+                list(dependent_vars) +
+                list(dependent_states)
+            )
+
+        try:
```

Comment:
It's probably a good idea to check if both models are valid at the start of this method, e.g.

        if not model.is_valid():
            model.validate()
            
Then you can just use .time() without the try-catch because a valid model always has a time variable

Reply:
Only downside is it then won't work for invalid models, but that's probably something we can live with?

Reply:
I was thinking there might be users who want to create a full model out of incomplete models and so wouldn't have fully valid models to input. Is this bit of code problematic or is it acceptable to leave like this to allow invalid models?

Reply:
Good point!